pycsw/ogc/csw/csw3.py

```diff
@@ -2049,9 +2053,9 @@ def resolve_nsmap(self, list_):
             LOGGER.debug(tn)
             if tn.find(':') != -1:  # resolve prefix
                 prefix = tn.split(':')[0]
-                if prefix in nsmap.keys():  # get uri
+                if prefix in list(nsmap.keys()):  # get uri
                     uri = nsmap[prefix]
-                    newprefix = self.parent.context.namespaces.keys()[self.parent.context.namespaces.values().index(uri)]
+                    newprefix = list(self.parent.context.namespaces.keys())[list(self.parent.context.namespaces.values()).index(uri)]
```

Comment:
Since there's no order to dictionary iteration, I don't think you should rely on `keys()` and `values()` returning lists in the same order. This should probably be re-written like the one a little bit above I commented on earlier.